### PR TITLE
[arch-v2][MED] Expand frontend test breadth (pages/components) + honest coverage

### PR DESCRIPTION
**Pull request by omniscient**
## Summary
Automated implementation for issue #198.

## Preview
_No preview environment — this change does not affect the running app ('All 16 changed files are test files, test utilities, or test configuration. No runtime application code is touched.')._

## Commands
```bash
# Iterate after feedback
docker compose --profile factory run --rm dark-factory "Continue issue #198"

# Tear down preview when done
docker compose --profile factory run --rm dark-factory "Close issue #198"
```

---
*Generated by MarketHawk Dark Factory*

### CODE REVIEW

File: frontend/src/components/QualityReportModal.test.tsx [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+  });
+
+  it('renders the grade badge after data loads', async () => {
+    renderWithQuery(
+      <QualityReportModal isOpen={true} onClose={vi.fn()} universe={mockUniverse} />
+    );
+    await waitFor(() => {
+      expect(screen.getAllByText('A').length).toBeGreaterThan(0);
+    });
+  });
```
Comment (omniscient) [inline frontend/src/components/QualityReportModal.test.tsx:69]: **edge-case** · **low** — `fireEvent.click(screen.getAllByRole('button')[0])` asserts onClose fires from whatever button happens to render first; if button order changes the test passes for the wrong reason. Query the close control by name/aria-label instead.

File: frontend/src/pages/Alerts/Alerts.test.tsx [hunk 2]
```diff
@@ -0,0 +1,48 @@
+import { vi, describe, it, expect, beforeAll } from 'vitest';
+import { screen } from '@testing-library/react';
+import { renderWithQuery } from '../../test-utils/renderWithQuery';
+import Alerts from './index';
+
+// ChannelConfigPanel reads Notification.permission at render time; jsdom doesn't define it
+beforeAll(() => {
+  Object.defineProperty(globalThis, 'Notification', {
+    value: { permission: 'default' },
```
Comment (omniscient) [inline frontend/src/pages/Alerts/Alerts.test.tsx]: **maintainability** · **low** — The `globalThis.Notification` polyfill is defined locally; any other test rendering `ChannelConfigPanel` will hit the same jsdom gap. Move it to `src/test-setup.ts` to avoid duplication.

File: frontend/src/pages/Login/Login.test.tsx [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+    renderWithQuery(<Login />);
+    await waitFor(() => {
+      expect(screen.getByRole('button', { name: /sign in/i })).toBeInTheDocument();
+    });
+  });
+
+  it('shows username and password fields', async () => {
+    const { container } = renderWithQuery(<Login />);
+    await waitFor(() => screen.getByRole('button', { name: /sign in/i }));
+    expect(container.querySelector('input[type="text"]')).toBeInTheDocument();
```
Comment (omniscient) [inline frontend/src/pages/Login/Login.test.tsx]: **maintainability** · **low** — Selecting fields via `container.querySelector('input[type="text"]')` couples the test to input type attributes; if a username field gains autocomplete/email type it silently breaks. Prefer `getByLabelText`/`getByRole('textbox')`.

File: frontend/vitest.config.ts [hunk 4]
```diff
@@ -9,19 +9,20 @@ export default defineConfig({
...
-        'src/hooks/useScannerState.ts',
-        'src/hooks/useScannerWs.ts',
-        'src/hooks/useScanTask.ts',
-        'src/hooks/useWatchlistLive.ts',
-        'src/hooks/useLiveStockData.ts',
-        'src/hooks/useScorecard.ts',
-        'src/components/ui/GlobalErrorToast.tsx',
+      all: true,
+      include: ['src/**'],
+      exclude: [
```
Comment (omniscient) [inline frontend/vitest.config.ts:14]: **maintainability** · **low** — `include: ['src/**']` with this exclude list still counts type-only/barrel files (e.g. `index.ts` re-exports, files that are pure interfaces) toward the denominator, depressing the "honest" number. Consider excluding obvious non-executable files (`src/**/types.ts`, index barrels) so the metric reflects testable code.

File: frontend/vitest.config.ts [hunk 5]
```diff
@@ -9,19 +9,20 @@ export default defineConfig({
...
+        'src/main.tsx',
+        'src/vite-env.d.ts',
+        'src/test-setup.ts',
+        'src/test-utils/**',
+        'src/**/*.test.{ts,tsx}',
+        'src/**/*.d.ts',
       ],
       thresholds: {
```
Comment (omniscient) [inline frontend/vitest.config.ts:25]: **maintainability** · **low** — The new whole-tree thresholds pass but only barely (functions 16.74% vs 15% floor; statements/branches/lines ~1-2pp of headroom). Any new uncovered source can break CI; consider setting floors a couple points below current actuals to absorb churn.

**Review by omniscient — COMMENTED** [review]
## 🏭 Dark Factory Code Review

**0** blocking · **5** advisory finding(s).

File: frontend/src/pages/Alerts/Alerts.test.tsx (hunk 2 above)
Comment (omniscient) [reply frontend/src/pages/Alerts/Alerts.test.tsx]: I agree to this change

File: frontend/src/pages/Login/Login.test.tsx (hunk 3 above)
Comment (omniscient) [reply frontend/src/pages/Login/Login.test.tsx]: Shoud be addressed
